dissect/target/plugins/filesystem/yara.py

```diff
@@ -16,48 +24,128 @@
         ("digest", "digest"),
         ("string", "rule"),
         ("string[]", "tags"),
+        ("string", "namespace"),
     ],
 )
 
+MAX_SCAN_SIZE = 10 * 1024 * 1024
 
-class YaraPlugin(Plugin):
-    """Plugin to scan files against a local YARA rules file."""
 
-    DEFAULT_MAX_SIZE = 10 * 1024 * 1024
+class YaraPlugin(InternalPlugin):
+    """Plugin to scan files against a local YARA rules file."""
 
     def check_compatible(self) -> None:
-        pass
+        if not HAS_YARA:
+            raise UnsupportedPluginError("Please install 'yara-python' to use the yara plugin.")
+
+    def yara(
+        self, rules: str | list[str | Path], path: str = "/", max_size: int = MAX_SCAN_SIZE, check_rules: bool = False
+    ) -> Iterator[YaraMatchRecord]:
+        """Scan files inside the target up to a given maximum size with YARA rule file(s)."""
+        rules_path = rules
+
+        if isinstance(rules, str):
+            rules = rules.split(",")
 
-    @arg("--rule-files", "-r", type=Path, nargs="+", required=True, help="path to YARA rule file")
-    @arg("--scan-path", default="/", help="path to recursively scan")
-    @arg("--max-size", "-m", default=DEFAULT_MAX_SIZE, help="maximum file size in bytes to scan")
-    @export(record=YaraMatchRecord)
-    def yara(self, rule_files, scan_path="/", max_size=DEFAULT_MAX_SIZE):
-        """Scan files up to a given maximum size with a local YARA rule file.
+        rules = process_rules(rules, check_rules)
 
-        Example:
-            target-query <TARGET> -f yara --rule-file /path/to/yara_sigs.rule
-        """
+        if not rules:
+            self.target.log.error("No working rules found in %s.", rules_path)
+            return set(())
 
-        rule_data = "\n".join([rule_file.read_text() for rule_file in rule_files])
+        if hasattr(rules, "warnings") and (num_warns := len(rules.warnings)) > 0:
+            log.warning("Yara generated %s warnings while compiling rules", num_warns)
 
-        rules = yara.compile(source=rule_data)
-        for _, _, files in self.target.fs.walk_ext(scan_path):
-            for file_entry in files:
-                path = self.target.fs.path(file_entry.path)
+        for _, _, files in self.target.fs.walk_ext(path):
+            for file in files:
                 try:
-                    if path.stat().st_size > max_size:
+                    if file.stat().st_size > max_size:
```

Comment:
Can you add a log line outside of this loop that says something along the lines of `Skipping files larger than ...`?